lib/matplotlib/testing/decorators.py

```diff
@@ -200,7 +205,16 @@ def remove_text(figure):
     def test(self):
         baseline_dir, result_dir = _image_directories(self._func)
 
-        for fignum, baseline in zip(plt.get_fignums(), self._baseline_images):
+        for fignum, baseline in zip_longest(plt.get_fignums(), self._baseline_images):
```

Comment:
Rather than using `izip_longest`, couldn't you just check that the lengths of `plt.get_fignums()` and `self._baseline_images` are the same before the loop?  That we we don't do a bunch of work only to fail later.  (And if either of those is an iterator, they are certainly very short iterators so the cost of doing `list()` on them should be negligible).